PlaywrigthUITests/PageObjects/DemoQAWebTablesPage.cs

```diff
@@ -89,16 +95,126 @@ public async Task VerifyTableRowContent(string headerName = "First Name", string
 
     public async Task VerifyPopupVisible()
     {
-        var popup = Page.Locator(".modal-content");
-        await Assertions.Expect(popup).ToBeVisibleAsync();
+        var isPopupOpened = await RegistrationPopup.IsVisibleAsync();
+        Assert.That(isPopupOpened, "The 'Add' popup is not visible.");
     }
 
     public async Task VerifyFirstNameVisible()
     {
-        var popup = Page.Locator(".modal-content");
-        var firstName = popup.GetByPlaceholder("First Name");
+        var firstName = Page.GetByPlaceholder("First Name");
         await Assertions.Expect(firstName).ToBeVisibleAsync();
     }
     
+    public async Task OpenAddPopup()
+    {
+        await AddButton.ClickAsync();
+        Assert.That(await RegistrationPopup.IsVisibleAsync(), "The 'Add' popup is not visible.");
+    }
+    
+    public async Task FillFirstName (string firstName)
+    {
+        if(firstName == null) throw new Exception("First Name field should not be null.");
```

Comment:
ArgumentException was correct here. There was only one place where it was incorrect (it was used when not an argument is verified). In all other places there were exactly argument exceptions

Reply:
fixed